Add Ruby example

kaylareopelle opened the pull request:
During a recent OpenTelemetry spike, the Ruby agent team created an application that closely mirrors the example in New Relic's [Python OpenTelemetry Workshop](https://developer.newrelic.com/opentelemetry-masterclass/hands-on/workshop-python/). 

We'd like to share this example with the community.

alanwest commented:
👋 Heya @kaylareopelle! Thanks for adding a Ruby example. You got this in draft, are you still working it?

kaylareopelle commented:
Hi @alanwest! Thanks for pinging me! This is ready for review 😄

reese-lee commented:
Hey @kaylareopelle! Thank you so much for putting this Ruby example together, and for separating it out into the directories per our Get Started Guide example in Java! I just have a few notes:

1. I know you mentioned OTel Ruby only has support for traces atm; I haven't gotten a chance to run the example yet, but does the tracing instrumentation produce what you would see [here](https://docs.newrelic.com/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/get-started/opentelemetry-tutorial-java/#traces) in the screen shots for traces?

2. I did see that it collects the same attributes, but can you rename them to `fibonacci.n` and `fibonacci.result`? 

3. The last thing (which I spaced on mentioning) is that we'd like all the examples to use the same port; could you update it to port 8080? 

Thanks again!

kaylareopelle commented:
Hi @reese-lee! Circling back to this today! I've addressed your feedback here: 
* Set default port to 8080: ca5a33e5b6a75e8c9552a0b4e4b052bb540a4fe4 
* Use `fibonacci.n` and `fibonacci.result` as attribute names: 6a8a2987d76dea814797b0d541b452863fa75233 


Here's a Ruby-equivalent of the screenshot you shared: 
<img width="1501" alt="Screenshot 2023-06-07 at 4 30 20 PM" src="https://github.com/newrelic/newrelic-opentelemetry-examples/assets/87386821/e94b5152-7836-4906-9527-21153d9465c5">

Please let me know if you need anything else!

kaylareopelle commented on README.md:
```diff
@@ -55,6 +55,8 @@ See [Contribute](#contribute) for how to request OpenTelemetry/New Relic interac
 - Go
   - [OpenTelemetry SDK New Relic Config](./go): Simple Go applications demonstrating OpenTelemetry instrumentation.
+- Ruby
+  - [OpenTelemetry SDK New Relic Config](./ruby): A Sinatra application with auto- and manual-instrumentation.
```
```suggestion
  - [OpenTelemetry SDK New Relic Config](./ruby): A Sinatra application with automatic and manual instrumentation.
```

kaylareopelle commented on getting-started-guides/ruby/instrumented/README.md:
````diff
@@ -0,0 +1,44 @@
+export OTEL_SERVICE_NAME=getting-started-ruby
+```
+
+4. Install your gems using Bundler
````
```suggestion
4. Install your gems using Bundler:
```

kaylareopelle commented on getting-started-guides/ruby/instrumented/README.md:
````diff
@@ -0,0 +1,44 @@
+bundle install
+```
+
+5. Run the following command
````
```suggestion
5. Run the following command:
```

kaylareopelle commented on getting-started-guides/ruby/instrumented/README.md:
````diff
@@ -0,0 +1,44 @@
+bundle exec rackup
+```
+
+4. To generate traffic, in a new terminal tab run the following command
````
```suggestion
4. To generate traffic, in a new terminal tab run the following command:
```

kaylareopelle commented on getting-started-guides/ruby/instrumented/README.md:
````diff
@@ -0,0 +1,44 @@
+```
+
+4. Install your gems using Bundler
+
````
```suggestion
```

kaylareopelle commented on getting-started-guides/ruby/instrumented/README.md:
````diff
@@ -0,0 +1,44 @@
+```
+
+5. Run the following command
+
````
```suggestion
```

reese-lee commented on getting-started-guides/ruby/instrumented/README.md:
```diff
@@ -0,0 +1,42 @@
+# Getting Started Guide - Ruuby
```
```suggestion
# Getting Started Guide - Ruby
```

reese-lee reviewed (approved):
Hey Kayla! This looks pretty good to me. Unfortunately, I ran into issues with gems on my machine so I have not been able to run the app myself; if you can confirm that the trace data the app produces matches the trace data we're looking for [here](https://docs.newrelic.com/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/get-started/opentelemetry-tutorial-java/#view-data), I am good to get this merged!

kaylareopelle commented:
@reese-lee - Sorry to hear you ran into trouble with gems! Here's a permalink to an app on staging with data from the instrumented app: https://staging.onenr.io/06vjAqpDnjP 

Things look quite similar. The only difference I see involves traces with errors: https://staging.onenr.io/0znQxBDeBRV 
The Ruby application doesn't have a nested span in this scenario. It has a nested span when the request completes successfully: https://staging.onenr.io/0PLREPoXewa

reese-lee commented:
@kaylareopelle I think that's good, that might just be how it's handled in Ruby.

kaylareopelle commented:
Great! In that case, please merge as you're able! (I don't have permission to do so)

kaylareopelle commented:
Hi @reese-lee! Sorry about the delay here. I think that makes sense. Is this what you're looking for? https://staging.onenr.io/0kERzpxLlQr
